pipeline.py

```diff
@@ -48,19 +59,32 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        pathlib_path_to_data = pathlib.Path(self.path_to_raw_txt_data)
+
+        for full_file_name in pathlib_path_to_data.iterdir():
```

Comment:
very strange loop variable naming. it might look like it is already a name, but in reality it is a Path instance, is not it?